WeakAuras/RegionTypes/Icon.lua

```diff
@@ -365,6 +378,8 @@ local function modify(parent, region, data)
 
   cooldown:SetReverse(not data.inverse);
   cooldown:SetHideCountdownNumbers(not data.cooldownTextEnabled or IsAddOnLoaded("OmniCC") or false);
+  cooldown:SetDrawSwipe(data.cooldownSwipe)
```

Comment:
Instead of calling SetDrawSwipe here directly, can you use the region:SetCooldownSwipe method you wrote?

Reply:
i could, but that would be inconsistent with the above code that doesn't use SetInverse either.

Reply:
Well, the region methods are inconsistent with themselves about this throughout the icon region. I’d like to move towards constructing the region methods, and then using them for initial setup. I think this will lead to greater consistency, and fewer opportunities for unexpected bugs to occur because some housekeeping was forgotten – just rely on the method that you just constructed to do the housekeeping.

Reply:
So, to be clear, the pattern I want is:

1. make region method
2. call region method with value from data
3. conditions call region methods with the appropriate values to update the region